### PR TITLE
Added support for neptune logger

### PR DESCRIPTION
**Pull request by harishankar-gopalan**
# What does this PR do ?

Adds Neptune logging support to NeMo which is already exposed by the PTL trainer.

**Collection**: Core, NLP

# Changelog 
- In experiment manager, consumed NeptuneLogger parameters from the OmegaConf
- Added tests for NeptuneLogger
- Added documentation for NeptuneLogger configuration

# Usage
* Can be directly used by configuring the mentioned keys in the docs under exp_manager similar to how WandB params are specified.

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

@titu1994 

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: tests/core/test_exp_manager.py [hunk 1]
```diff
@@ -195,6 +195,36 @@ def test_trainer_loggers(self, tmp_path):
         )
         assert isinstance(test_trainer.logger, pl.loggers.WandbLogger)
 
+    @pytest.mark.unit
+    def test_trainer_neptune_logger(self, tmp_path):
+        pytest.importorskip("neptune", reason="could not import `neptune`, use `pip install neptune` to run this test")
```
Comment (harishankar-gopalan) [inline tests/core/test_exp_manager.py:200]: Added conditional test case skip if neptune is not installed.

File: nemo/utils/exp_manager.py [hunk 2]
```diff
@@ -872,6 +879,20 @@ def configure_loggers(
...
 
+    if create_neptune_logger:
+        if neptune_kwargs is None:
+            neptune_kwargs = {}
+        if "name" not in neptune_kwargs and "project" not in neptune_kwargs:
+            raise ValueError("name and project are required for neptune_logger")
+        if "api_key" not in neptune_kwargs and not os.getenv("NEPTUNE_API_TOKEN", None):
+            raise ValueError(
+                "either api_key should be set in neptune_kwargs or NEPTUNE_API_TOKEN should be set in environment variable for neptune_logger"
+            )
```
Comment (harishankar-gopalan) [inline nemo/utils/exp_manager.py:890]: Havent checked for import error here, as PyTorchLightning by default checks for the same within its NeptuneLogger wrapper.

**Comment by titu1994** [comment]
jenkins

**Comment by titu1994** [comment]
jenkins

**Comment by harishankar-gopalan** [comment]
Hi @titu1994 what would be the next process to get this merged to the main trunk ?

**Comment by titu1994** [comment]
Could you avoid merging the main branch with this PR, I need to see the tests pass. It is fine if the branch is out of date with the main branch.

**Comment by titu1994** [comment]
jenkins

**Comment by harishankar-gopalan** [comment]
> jenkins

I guess the automation is completed and all checks have passed.

**Review by titu1994 — APPROVED** [review]
Thanks for the PR !

**Pull request merged**